src/source/CppMarshal.scala

```diff
@@ -6,27 +6,38 @@ import djinni.meta._
 
 class CppMarshal(spec: Spec) extends Marshal(spec) {
 
-  override def typename(tm: MExpr): String = toCppType(tm, None)
+  override def typename(tm: MExpr): String = toCppType(tm, Seq(), None)
+  def typename(tm: MExpr, scopeNames: Seq[String]): String = toCppType(tm, scopeNames, None)
+  def typename(ty: TypeRef, scopeNames: Seq[String]): String = typename(ty.resolved, scopeNames)
   def typename(name: String, ty: TypeDef): String = ty match {
     case e: Enum => idCpp.enumType(name)
     case i: Interface => idCpp.ty(name)
     case r: Record => idCpp.ty(name)
   }
 
-  override def fqTypename(tm: MExpr): String = toCppType(tm, Some(spec.cppNamespace))
+  override def fqTypename(tm: MExpr): String = toCppType(tm, Seq(), Some(spec.cppNamespace))
```

Comment:
Sure. I apparently can't add a default argument to the second `toCppType()` because the Scala compiler complains. I'm adding one for `toCppParamType()`.